### PR TITLE
Add edit icon and assessment side-bar to areas

### PR DESCRIPTION
**Pull request by NingyueZhou**
If users are logged in, they will see an edit icon in area card, clicking on it will open a sidebar for assessment. 
The content of sidebar is empty because I discussed with @mourabitiziyad. Implementing the assessment content will be double work as he is working on it.
image
image

### CODE REVIEW

File: src/components/ui/areaCard.tsx [hunk 1]
```diff
@@ -21,16 +35,89 @@ const AreaCard = React.forwardRef<HTMLDivElement, AreaProps>((area, ref) => {
...
         <SheetTrigger>
-          <div className="h-4 w-4 opacity-50 shrink-0 rounded-full hover:bg-accent hover:text-accent-foreground">
+          <div className="h-4 w-4 shrink-0 rounded-full opacity-50 hover:bg-accent hover:text-accent-foreground">
             <Icons.info />
           </div>
         </SheetTrigger>
+        <SheetContent className="lg:min-w-[40vw]">
+          <SheetHeader>
+            <SheetTitle className="text-center">
+              <div>Area Details: </div>
```
Comment (severynenko) [inline src/components/ui/areaCard.tsx]: Maybe it would make sense to have a generic Sheet content component with these details structure if it is reused

**Review by mourabitiziyad — CHANGES_REQUESTED** [review]
I think the implementation was made a lot more complicated than it should have. 

I believe the implementation should be just a basic empty `<Sheet />` component imported into the Refa with the included `<SheetTrigger />` be the edit icon. 

You can keep the logic of only displaying the edit button trigger when the user is logged in. But I think there is no need for wrapping/augmenting logic onto the existing components that handle Refa.

So I would personally expect this PR to include only two file changes max.

**Comment by NingyueZhou** [comment]
> I think the implementation was made a lot more complicated than it should have.
> 
> I believe the implementation should be just a basic empty `<Sheet />` component imported into the Refa with the included `<SheetTrigger />` be the edit icon.
> 
> You can keep the logic of only displaying the edit button trigger when the user is logged in. But I think there is no need for wrapping/augmenting logic onto the existing components that handle Refa.
> 
> So I would personally expect this PR to include only two file changes max.

@mourabitiziyad If I understood correctly, you would suggest only having `areaCard` instead having two components `areaCard` and `areaCardAssessment`, and just add new `<Sheet />` and `<SheetTrigger />` into existing `areaCard` component, right?
My thought here creating a new component `areaCardAssessment` is that in this component we can add more assessment logic later on in this assessment specific logic, instead of having everything in `areaCard` and distinguish unlogged in status and assessment logic within one file. What would be your suggestion on this? Maybe you see a different way of implementing it

**Comment by NingyueZhou** [comment]
Improved implementation according to review comments, only minor changes in `areaCard`, simplified `areaCardAssessment` implementation, no doubled area information sheet content

2 extra file changes just due to renaming from `addnewprojectcard` to `addNewProjectCard`

**Comment by mourabitiziyad** [comment]
> Improved implementation according to review comments, only minor changes in `areaCard`, simplified `areaCardAssessment` implementation, no doubled area information sheet content
> 
> 2 extra file changes just due to renaming from `addnewprojectcard` to `addNewProjectCard`

Looks much better already! Thanks :D 

I will have a proper look after the meeting today :)

File: src/components/ui/areaCardAssessment.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+              size="icon"
+              className="h-4 w-4 shrink-0 rounded-full opacity-50 hover:bg-accent hover:text-accent-foreground"
+            >
+              <Icons.info />
+            </Button>
+          </SheetTrigger>
+
+          {/* Edit icon to trigger assess sheet */}
+          <Sheet>
+            <SheetTrigger>
```
Comment (mourabitiziyad) [inline src/components/ui/areaCardAssessment.tsx]: SheetTrigger is already a button, putting a button inside of a button creates Hydration errors.

File: src/components/icons.tsx [hunk 3]
```diff
@@ -373,4 +373,26 @@ export const Icons = {
...
+  edit: (props: IconProps) => (
+    <svg
+      width="15"
+      height="15"
+      viewBox="0 0 15 15"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <path
+        fill-rule="evenodd"
```
Comment (mourabitiziyad) [inline src/components/icons.tsx]: fill-rule in React should be FillRule, same for the other props.

File: src/components/ui/areaCardAssessment.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  SheetDescription,
+  SheetHeader,
+  SheetTitle,
+  SheetTrigger,
+} from "~/components/ui/sheet";
+import { Card, CardContent, CardHeader, CardTitle } from "./card";
+import { Button } from "./button";
+import { Icons } from "../icons";
+import { Badge } from "~/components/ui/badge";
+import { Separator } from "~/components/ui/separator";
```
Comment (mourabitiziyad) [inline src/components/ui/areaCardAssessment.tsx]: Try not to skip these unused imports

File: src/components/ui/areaCard.tsx [hunk 5]
```diff
@@ -21,15 +21,19 @@ const AreaCard = React.forwardRef<HTMLDivElement, AreaProps>((area, ref) => {
...
         <SheetTrigger>
-          <div className="h-4 w-4 opacity-50 shrink-0 rounded-full hover:bg-accent hover:text-accent-foreground">
+          <Button
+            variant="outline"
+            size="icon"
+            className="h-4 w-4 shrink-0 rounded-full opacity-50 hover:bg-accent hover:text-accent-foreground"
+          >
             <Icons.info />
-          </div>
+          </Button>
```
Comment (mourabitiziyad) [inline src/components/ui/areaCard.tsx]: why did you put a button inside SheetTrigger? SheetTrigger is already a button

File: src/components/ui/cardWrapper.tsx [hunk 6]
```diff
@@ -71,12 +75,22 @@ const CardWrapper = React.forwardRef<HTMLDivElement, CardWrapperProps>(
...
+            />
+          ) : (
+            <AreaCard
+              id={area.id}
+              name={area.name}
+              visible={isAreaVisible}
+              toggleVisibility={toggleVisibility}
+            />
+          )}
         </div>
```
Comment (mourabitiziyad) [inline src/components/ui/cardWrapper.tsx]: This logic is not needed. You are displaying the two first buttons anyway, why have two components that share the same buttons? Always show AreaCard, and if the user is logged in, show the edit button as well. No need to create a component that is identical to AreaCard and augment it with the edit button.

If we wish to edit the info button for instance, we'll now have to do it twice, which is redundant.

**Review by mourabitiziyad — CHANGES_REQUESTED** [review]

File: src/components/ui/areaCardAssessment.tsx (hunk 2 above)
Comment (NingyueZhou) [reply src/components/ui/areaCardAssessment.tsx]: Will remove the nested button.

One thing I noticed: the button of `<Button>` component looks different from the button of `<SheetTrigger>`, see screenshots here (the button of `<Button>` component has a slightly gray background)
image
If a button inside of sheettrigger, they look the same
image

But of course, will remove button inside of sheettrigger, thank you for point it out

File: src/components/ui/areaCard.tsx (hunk 5 above)
Comment (NingyueZhou) [reply src/components/ui/areaCard.tsx]: Will also remove this

File: src/components/ui/cardWrapper.tsx (hunk 6 above)
Comment (NingyueZhou) [reply src/components/ui/cardWrapper.tsx]: I see your point, let me change it again

File: src/components/icons.tsx [hunk 7]
```diff
@@ -382,14 +382,14 @@ export const Icons = {
...
+        fillRule="evenodd"
+        clipRule="evenodd"
         d="M10.3409 5.71455L9.28544 4.65908L2.83112 11.1134L2.32375 12.6762L3.88659 12.1689L10.3409 5.71455ZM4.68853 13.4779L0 15L1.52212 10.3115L9.28544 2.54814L12.4518 5.71455L4.68853 13.4779Z"
         fill="#6A7181"
       />
       <path
-        fill-rule="evenodd"
-        clip-rule="evenodd"
+        fillRule="evenodd"
+        clipRule="evenodd"
```
Comment (mourabitiziyad) [inline src/components/icons.tsx:392]: Fixed incorrect prop spelling from `FillRule` to `fillRule`

**Pull request merged**